src/generators/functiongenerators/CustomDistribution.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (C) 2024 European Spallation Source, ERIC. See LICENSE file
+//===----------------------------------------------------------------------===//
+///
+/// \file
+///
+/// \brief Creates a custom TOF distribution
+//===----------------------------------------------------------------------===//
+// GCOVR_EXCL_START
+
+#pragma once
+
+#include <random>
+
+class CustomDistribution {
```

Comment:
I would call this FunctionalDistribution or something like this since it can give distribution from a function